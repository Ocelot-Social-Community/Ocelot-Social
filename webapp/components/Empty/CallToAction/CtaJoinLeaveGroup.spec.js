import { mount, RouterLinkStub } from '@vue/test-utils'

import Component from './CtaJoinLeaveGroup.vue'

const localVue = global.localVue

describe('CtaJoinLeaveGroup.vue', () => {
  let propsData, wrapper, mocks, stubs

  beforeEach(() => {
    propsData = {
      group: {
        id: 'g-123',
        slug: 'group-123',
        name: 'Group 123',
        myRole: null,
      },
    }
    mocks = {
      $t: jest.fn((t) => t),
      $store: {
        getters: {
          'auth/user': {
            id: 'u-1',
          },
        },
      },
      $apollo: {
        mutate: jest.fn(),
      },
    }
    stubs = {
      NuxtLink: RouterLinkStub,
    }
  })

  const Wrapper = () => {
    return mount(Component, { propsData, localVue, mocks, stubs })
  }

  describe('mount', () => {
    beforeEach(() => {
      wrapper = Wrapper()
    })

    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('clicking on button', () => {
      beforeEach(async () => {
        mocks.$apollo.mutate = jest.fn().mockResolvedValue({
          data: {
            JoinGroup: {
              id: 'g-123',
              slug: 'group-123',
              name: 'Group 123',
              myRoleInGroup: 'usual',
            },
          },
        })
        wrapper.find('.base-button').trigger('click')
        await wrapper.vm.$nextTick()
      })
      it('emits update event', async () => {
        expect(wrapper.emitted().update).toEqual([
          [
            {
              id: 'g-123',
              slug: 'group-123',
              name: 'Group 123',
              myRoleInGroup: 'usual',
            },
          ],
        ])
      })
    })
  })
})
