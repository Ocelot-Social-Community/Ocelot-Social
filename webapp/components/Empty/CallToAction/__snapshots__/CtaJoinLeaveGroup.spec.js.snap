// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaJoinLeaveGroup.vue mount renders 1`] = `
<div class="ds-space ds-space-centered" style="margin-top: 2px; margin-bottom: 2px;">
  <div class="ds-space" style="margin-bottom: 16px;"></div>
  <h4 class="ds-heading ds-heading-h4">
    contribution.comment.commenting-disabled.no-group-member.reason
    <a>
      Group 123
    </a>
  </h4>
  <p class="ds-text">
    contribution.comment.commenting-disabled.no-group-member.call-to-action
  </p> <button type="button" class="join-leave-button base-button --filled has-tooltip" data-original-title="null"><span class="base-icon"><!----></span>
    <!---->
    group.joinLeaveButton.join
  </button>
</div>
`;
